src/Umbraco.Web/WebApi/Filters/AngularAntiForgeryHelper.cs

```diff
@@ -111,15 +107,41 @@ public static bool ValidateTokens(string cookieToken, string headerToken)
         /// <returns></returns>
         public static bool ValidateHeaders(HttpRequestHeaders requestHeaders, out string failedReason)
         {
-            var cookieToken = requestHeaders
-                .GetCookies()
-                .Select(c => c[CsrfValidationCookieName])
-                .FirstOrDefault();
+            var cookieToken = GetCookie(requestHeaders, CsrfValidationCookieName);
 
             return ValidateHeaders(
                 requestHeaders.ToDictionary(x => x.Key, x => x.Value).ToArray(),
-                cookieToken == null ? null : cookieToken.Value,
+                cookieToken == null ? null : cookieToken,
                 out failedReason);
         }
+
+        /// <summary>
+        /// Retrieves an individual cookie from the cookies collection
+        /// Adapted from: https://stackoverflow.com/a/29057304/5018
+        /// </summary>
+        /// <param name="requestHeaders"></param>
+        /// <param name="cookieName"></param>
+        /// <returns></returns>
+        private static string GetCookie(HttpRequestHeaders requestHeaders, string cookieName)
+        {
+            var cookieRequestHeader = requestHeaders.Where(h => h.Key.Equals("Cookie", StringComparison.InvariantCultureIgnoreCase)).ToArray();
```

Comment:
This should be casted to ToArray and then use it's Length to verify if there is any. Linq has become far too overused and causes re-iteration of collections and is slower since anytime you use Linq it's going to allocate an Enumerator object. In some cases of course it makes sense but in tons of cases, creating an Array and using an Array as it was meant to be is much better. 

So after this it would be `if (cookieRequestHeader.Length == 0)` instead of the `Any()` check

There's also no need to then do `cookieRequestHeader.FirstOrDefault();` since you are already checking the length, you'd just do `var cookiesHeader = cookieRequestHeader[0];` and there's no need for a null check since we've already verified the collection contains what we're looking for.

I'll make some updates